### DISCUSSION TITLE
WorkerThreads in AVA

### DISCUSSION THREAD

**Discussion by AkbarHabeeb** [avajs/ava#3337] [Q&A]
Hi @novemberborn and AVA team.
**Context:** My project uses worker threads to log the details, when I migrate from V2 to V5 the tests are not working as expected. However they worked after adding a configuration _"workerThreads": false_

1. Why we need to disable the workedThreads in AVA?

Im interested in knowing the details indepth, please enlight me

**Comment by novemberborn** [comment]
I'm not sure if you can create a worker thread from inside a worker thread actually. Child processes may be better.

**Reply by AkbarHabeeb** [reply]
thanks for your reply

Im trying to run the test in V5 AVA with workerThreads disabled but then getting the below error. Btw the service uses worker threads for logging.
 ```
 TypeError: Cannot read properties of undefined (reading 'paths')

  › module.exports.internals.Monitor.start (node_modules/@hapi/good/lib/monitor.js:100:108)
  › Object.register (node_modules/@hapi/good/lib/index.js:73:17)
  › internals.Server.register (node_modules/@hapi/hapi/lib/server.js:494:35)
  › module.exports (node_modules/logger-lib/lib/logging/worker-threads.js:10:14)
  › TestApp._initializeBootLogger (node_modules/service-lib/lib/init.js:91:12)
```

Why we are not able to run the tests in V5 however in AVA V2 its working fine? Any help on this will be useful @novemberborn

**Comment by AkbarHabeeb** [comment]
@novemberborn @sindresorhus If workerThreads has been configured in ava as false, then is it possible and valid for me to create worker thread in the lib code?

Your answers will really help us here

**Reply by novemberborn** [reply]
I'd guess so, give it a try.

**Comment by AkbarHabeeb** [comment]
@novemberborn @sindresorhus I have one more question. In AVA V5 if the workerThreads has been set as false, do AVA still run the tests in a single thread or it will use subprocesses? The reason Im asking this was, when we try to register a hapi.js plugin in worker thread ... AVA was not allowing to do it since not all properties are set... which will be set if we have the correct context. From the research from chatgpt discussion, the properties wont be set if they are started by worker threads

**Reply by sindresorhus** [reply]
If `workerThreads` is disabled, then test files are run in separate child processes.
